src/components/Experiment.vue

```diff
@@ -83,20 +117,47 @@ export default {
       }
       this.currentScreen += 1;
     },
+    /**
+     * Add a result set
+     * @public
+     * @param data{Object} a flat object whose data you want to add to the results
+     */
     addResult(data) {
       if (!this.results[this.currentScreen]) {
         this.results[this.currentScreen] = [];
       }
       this.results[this.currentScreen].push(data);
-    },
-    getRootElement() {
-      return this.$el;
     }
+  },
+  /**
+   * The contents of this slot will be visible during the entire experiment
+   * @slot title
+   */
+  /**
+   * Place your screens inside this slot. They will be visible one after the other, like a slide show.
+   * @slot screens
+   */
+  render(h) {
+    // HACKY-O
+    this.$parent.$exp = this;
```

Comment:
Would there be another way to do this? I think the classic idea for Vue is "props down events up". e.g. instead of having the child directly modify anything on the parent, the child should be agnostic of what the parent is. Instead of adding a button besides each screen saying 

```
        <button
          @click="
            $exp.addResult({ answer });
            $exp.nextScreen();
          "
        >
          Next
        </button>
```

Would it be possible to let each Screen emit an event upwards and let the parent then decide how to handle that event.

Reply:
Also, what would actually be the `$parent` of `Experiment`. I thought `Experiment` is already the element at root level?

Reply:
Now I see that you have all the trial components emit an `answer:change` event upon answer changes, which makes sense. I guess I was just wondering if we could integrate the "next" button also in the screen component or even the trial components, instead of having to add it manually every time besides a screen.

Reply:
> what would actually be the $parent of Experiment. 

the $parent of Experiment is the App component. The reason for injecting `$exp` as hackily as seen here, was that I wanted `$exp` to be available everywhere, including the App component. I see $exp as something similar to vue router, which is also available on every component.

> I guess I was just wondering if we could integrate the "next" button also in the screen component or even the trial components, instead of having to add it manually every time besides a screen.

This implementation for provides flexibility. You can either add a button or you can save and switch screens immediately on change, allowing implementors to aggregate multiple inputs in one screen.